### DISCUSSION TITLE
issue: i want to calculate row data i will give quantity & price should calculate subtotal

### DISCUSSION THREAD

**Discussion by NarenderDhamarlapally** [react-hook-form/react-hook-form#11161] [Q&A]
### Version Number

7.14.1

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-v7-form-context-forked-7t7dfp?file=/src/index.js

### Steps to reproduce

1. Go to 'codesandbox'
2. but it is rendering multiple loops why because of usewatch
3. Click on quantity and price should populate subtotal data  
5. See error

### Expected behaviour

when quantity and price enter should populate subtotal data 
and also add 

  // (prevProps, nextProps) =>
  // prevProps.formState.isDirty === nextProps.formState.isDirty
  
  please add this is Test component.
prevProps.formState is getting undefined 
 nextProps.formState is getting undefined

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by NarenderDhamarlapally** [comment]
bill please help me for this .
this is the main part of my application calculation

**Comment by mutasim77** [comment]
This line is causing a re-render of the component, and it's becoming an infinite loop:
```js
setValue(`lineItems[${index}][subtotal]`, output);
```

To fix this issue, you can use the useEffect hook to update the value after the component has rendered:
```js
import React, { useEffect } from "react";
import { useWatch } from "react-hook-form";

export const Calc = ({ control, setValue, index }) => {
  const results = useWatch({ control, name: `lineItems[${index}]` });

  const output = results?.quantity * results.price;

  useEffect(() => {
    setValue(`lineItems[${index}].subtotal`, output);
  }, [index, output, setValue]);

  return <p>{output}</p>;
};
```

**Reply by NarenderDhamarlapally** [reply]
https://codesandbox.io/s/react-hook-form-v7-form-context-forked-c9lpvw?file=/src/Calc.js

first time data is coming from next time data is not changed
